pkg/autodiscovery/config_poller.go

```diff
@@ -21,7 +22,8 @@ import (
 // `ConfigProvider` and whether it should be polled or not.
 type configPoller struct {
 	provider     providers.ConfigProvider
-	configs      map[uint64]*integration.Config
+	configs      map[uint64]integration.Config
```

Comment:
I've been curious why AD always handles integration.Config's by-value -- it's quite a large struct to be copying all over the place!

Reply:
I do wonder the same!